pkg/systemd/reporting.jsx

```diff
@@ -307,184 +319,124 @@ class BusWorkflowRow extends React.Component {
                             return;
                         }
 
-                        console.error(cockpit.format("reportd task for workflow $0 did not finish: $1", this.props.workflow[0], (ex.problem || ex.message)));
-                        this.setState({ message: _("Reporting failed") });
+                        console.error(cockpit.format("reportd task for workflow $0 did not finish: $1", workflow[0], (ex.problem || ex.message)));
+                        setMessage(_("Reporting failed"));
                     });
                 })
                 .catch(ex => console.error(cockpit.format("Setting up a D-Bus proxy for $0 failed: $1", object_path, ex)));
-    }
-
-    _onReportButtonClick(event) {
-        this.setState({ problemState: ProblemState.UNREPORTABLE });
+    };
 
-        this.setState({
-            message: _("Waiting to start…"),
-            problemState: ProblemState.REPORTING,
-        });
+    const onReportButtonClick = (_event) => {
+        setProblemState(ProblemState.UNREPORTABLE);
+        setMessage(_("Waiting to start…"));
+        setProblemState(ProblemState.REPORTING);
```

Comment:
This looks an issue coming from the old code, but why change the `problemState` state twice in a row? I think React will batch the `setProblemState` calls and only one of them (the last one) will take effect.

Reply:
Yes, good point, we should just drop `UNREPORTABLE`